dss_datamover/tests/conftest.py

```diff
@@ -0,0 +1,182 @@
+# CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+# NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+# ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""
+
+from utils import utility, config
+from logger import MultiprocessingLogger
+from master_application import Master, process_put_operation, process_list_operation, process_get_operation, process_del_operation
+from multiprocessing import Queue, Value, Lock
+
+import json
+import os
+import pytest
+import tempfile
+
+
+@pytest.fixture(scope="session")
+def get_pytest_configs():
+    pytest_config_filepath = os.path.dirname(__file__) + "/pytest_config.json"
+    with open(pytest_config_filepath) as f:
+        pytest_configs = json.load(f)
+    return pytest_configs
+
+
+@pytest.fixture(scope="session")
+def get_config_object():
+    test_config_filepath = os.path.dirname(__file__) + "/pytest_config.json"
+    config_obj = config.Config({}, config_filepath=test_config_filepath)
+    return config_obj
+
+
+@pytest.fixture(scope="session")
+def get_config_dict(get_config_object):
+    return get_config_object.get_config()
+
+
+@pytest.fixture(scope="session")
+def get_system_config_object():
+    config_obj = config.Config({}, config_filepath="/etc/dss/datamover/config.json")
+    return config_obj
+
+
+@pytest.fixture(scope="session")
+def get_system_config_dict(get_system_config_object):
+    return get_system_config_object.get_config()
+
+
+@pytest.fixture
+def get_multiprocessing_logger(tmpdir):
+    logger_status = Value('i', 0)  # 0=NOT-STARTED, 1=RUNNING, 2=STOPPED
+    logger_queue = Queue()
+    logger_lock = Lock()
+    logging_path = tmpdir
+    logging_level = "INFO"
+
+    logger = MultiprocessingLogger(logger_queue, logger_lock, logger_status)
+    logger.config(logging_path, __file__, logging_level)
+    logger.start()
+
+    yield logger
+
+    # teardown logger
+    logger.stop()
+
+
+@pytest.fixture
+def clear_datamover_cache():
+    cache_files = ["/var/log/dss/prefix_index_data.json", "/var/log/dss/dm_resume_prefix_dir_keys.txt"]
+    for f in cache_files:
+        if os.path.exists(f):
+            os.remove(f)
+
+
+class MockSocket():
+    """
+    Dummy Object for an actual socket, should simulate all basic functions of a socket object
+    """
+    # TODO: finish building out MockSocket class
+    def __init__(self):
+        self.mockhost = "192.168.300.144"
+        self.mockport = "4000"
```

Comment:
Hardcoding IPs is not accepted. Someone has to remember to edit these files when they want to validate on their machines.

Reply:
this is a fake ip / port

Reply:
replaced with x chars instead